language_service/src/state.rs

```diff
@@ -414,7 +414,7 @@ impl<'a> CompilationStateUpdater<'a> {
                     .map(|manifest| manifest.lints)
                     .unwrap_or_default(),
             };
-            let configuration = merge_configurations(&notebook_configuration, &configuration);
+            let configuration = merge_configurations_for_notebook(&notebook_configuration, &configuration);
```

Comment:
this feels like overkill - couldn't you just have the `notebook_metadata.target_profile` default to Unrestricted above if it's not specified?

Actually, maybe just look at whether the VS Code layer (who constructs this notebook metadata) should just pass in Unrestricted as a default (if you go that route you can probably just revert your rust changes). Since really it's just VS Code business logic we're talking about here, so that deafult belongs there. You may want to add a VS Code integration test for that if you go that route